Guides/Overwrite.md

```diff
@@ -0,0 +1,124 @@
+# Overwrite
+
+[[Source](../Sources/Algorithms/Overwrite.swift) |
+ [Tests](../Tests/SwiftAlgorithmsTests/OverwriteTests.swift)]
+
+Copy a sequence onto an element-mutable collection.
+
+```swift
+var destination = Array("abcde")
+print(String(destination))  // "abcde"
+
+let source = "123"
+let changedEnd = destination.overwrite(prefixWith: source)
+print(String(destination))  // "123de"
+print(String(destination[changedEnd...]))  // "de"
+```
+
+`overwrite(prefixWith:)` takes a source sequence and replaces the first `k`
+elements of the receiver with the first `k` elements of the source, where *k*
+is the length of the shorter sequence. `overwrite(forwardsWith:)` does the same
+thing with a source collection, and `overwrite(prefixUsing:)` with an `inout`
+source iterator. To preserve memory exclusivity, the
+`overwrite(forwardsFrom:to:)` overload is required to copy between subsequences
+of the same collection, where the source and destination are given as index
+ranges.
+
+When the receiving element-mutable collection supports bidirectional traversal,
+variants of the previous methods are defined that copy the source elements on
+top of the receiver's suffix instead. The `overwrite(suffixWith:)` and
+`overwrite(suffixUsing:)` methods use their source's prefix, while the
+`overwrite(backwardsWith:)` and `overwrite(backwardsFrom:to:)` methods use
+their source's suffix.
+
+## Detailed Design
+
+New methods are added to element-mutable collections:
+
+```swift
+extension MutableCollection {
```

Comment:
This indeed seems a nice proposal and well implemented, but I was wondering that this functionality is kind of already covered in a more general way by `RangeReplaceableCollection`'s `replaceSubrange(:with:)`. 
So those are more questions/suggestions: 
What are the advantages of having this API over the existing one? 
And also, related to naming, could this complement the existing one by being called `replacePrefix(with:)`? 
I think this is more align with the convention and easier to reason about given the existing API's available.

Reply:
`RangeReplaceableCollection` and `MutableCollection` generally can’t be substituted for each other; the API you mentioned is pretty much their sole spiritual overlap, and it wouldn’t work with collections that implement only `MC`.